pkgs/stdenv/generic/default.nix

```diff
@@ -90,9 +90,14 @@ let
       '' + lib.optionalString hostPlatform.isDarwin ''
         export NIX_DONT_SET_RPATH=1
         export NIX_NO_SELF_RPATH=1
-      '' + lib.optionalString targetPlatform.isDarwin ''
-        export NIX_TARGET_DONT_SET_RPATH=1
-      '';
+      ''
+      # TODO this should be uncommented, but it causes stupid mass rebuilds. I
+      # think the best solution would just be to fixup linux RPATHs so we don't
+      # need to set `-rpath` anywhere.
+      # + lib.optionalString targetPlatform.isDarwin ''
+      #   export NIX_TARGET_DONT_SET_RPATH=1
```

Comment:
We don't really want rpaths on darwin, can we keep this for native darwin builds?

Reply:
Yeah due to the way cc-wrapper works, mapping these to the infix salt, it will also do `depsTargetTarget` (intended effect of `NIX_TARGET_DONT_SET_RPATH`) for native builds. Yay impurities!